source/image/Mask.cpp

```diff
@@ -434,19 +434,24 @@ const vector<vector<Point>> &Mask::Outlines() const
 
 
 
-Mask Mask::operator*(double scale) const
+Mask Mask::operator*(Point scale) const
 {
 	Mask newMask = *this;
+	newMask.radius = 0.;
 	for(auto &outline : newMask.outlines)
+	{
 		for(Point &p : outline)
 			p *= scale;
-	newMask.radius *= scale;
+		double radius = ComputeRadius(outline);
+		if(radius > newMask.radius)
+			newMask.radius = radius;
```

Comment:
Spoon feed me: Why's this needed?

Reply:
The damage calculations will be off otherwise - they use the radius as an estimate, and if the scale is changed it will still use the unscaled radius unless we update it.

Reply:
This particular routine just finds the distance to the furthest-out point in the collision mask.

Reply:
Ah, I see. I was reading it wrong before.